charts/localstack/templates/deployment.yaml

```
@@ -71,7 +71,11 @@ spec:
             {{- end }}
             {{- if .Values.persistence.enabled }}
               - name: data
+              {{- if or (eq .Values.image.tag "latest") (not (hasPrefix .Values.image.tag "0.")) }}
+                mountPath: /var/lib/localstack
+              {{- else }}
                 mountPath: /tmp/localstack
+              {{- end }}
```

[`hasPrefix`](https://helm.sh/docs/chart_template_guide/function_list/#hasprefix-and-hassuffix) takes the prefix as its first parameter, i.e. it should be `hasPrefix "0." .Values.image.tag`. And the check for latest is actually redundant, you could just check if it's a `0.x` version in the conditional:
```suggestion
              {{- if hasPrefix "0." .Values.image.tag }}
                mountPath: /tmp/localstack
              {{- else }}
                mountPath: /var/lib/localstack
              {{- end }}
```